src/Shared/Recruit.Vacancies.Client/Infrastructure/Services/EmployerAccount/EmployerAccountProvider.cs

```diff
@@ -85,5 +85,25 @@ public async Task<string> GetEmployerAccountPublicHashedIdAsync(long accountId)
                 throw;
             }
         }
+
+        public async Task<bool> GetEmployerAccountEOIAsync(string accountId)
+        {
+            try
+            {
+                var account = await _accountApiClient.GetAccount(accountId);
+                foreach (var entity in account.LegalEntities)
+                {
+                    var legalEntity = await _accountApiClient.GetLegalEntity(account.HashedAccountId, long.Parse(entity.Id));
+                    return legalEntity.Agreements.All(ag =>
+                        ag.AgreementType == SFA.DAS.Common.Domain.Types.AgreementType.NonLevyExpressionOfInterest);
+                }
+                return true;
```

Comment:
So if there are no legal entities. I would assume that would mean there are no signed EOI agreements and would expect this line to `return false;`